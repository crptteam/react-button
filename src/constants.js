export const SIZES = {
  large: 'large',
  normal: 'normal'
};
