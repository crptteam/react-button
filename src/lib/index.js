export {
  render,
  renderActive,
  renderNormal,
  renderDisabled,
  renderHovered,
  renderContent,
} from './renderUtils';
