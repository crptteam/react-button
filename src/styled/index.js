export { default as ButtonWrapper2 } from './ButtonWrapper';
export { default as Content } from './Content';
export { default as ButtonWrapper } from './ButtonWrap';