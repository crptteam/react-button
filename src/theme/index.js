import defaultTheme from './defaultTheme';
import filledTheme from './filledTheme';
import emptyTheme from './emptyTheme';
import whiteTheme from './whiteTheme';
import bgWhiteTheme from './bgWhiteTheme';

export { defaultTheme, filledTheme, emptyTheme, whiteTheme, bgWhiteTheme };
