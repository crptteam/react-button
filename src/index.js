import Button from "./components/Button";
import {SIZES} from "./constants";
import * as themes from "./theme";

export {
  Button,
  SIZES,
  themes
};

export default Button;