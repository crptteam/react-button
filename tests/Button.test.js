import React from "react";
import { Button } from "../src/";

describe("Button", () => {
  it("Should renders without problems", () => {
    const wrapper = shallow(<Button>Click me</Button>);

    expect(wrapper).toMatchSnapshot();
  });
});
