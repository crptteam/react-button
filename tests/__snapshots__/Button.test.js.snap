// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should renders without problems 1`] = `
<ButtonWrap
  innerRef={[Function]}
  onClick={[Function]}
  size="normal"
  theme={
    Object {
      "borderRadius": "2px",
      "disabled": Object {
        "background": "rgba(0,0,0,0)",
        "border": "1px solid #ABADB6",
        "color": "#ABADB6",
      },
      "fontSize": "14px",
      "fontWeight": 400,
      "large": Object {
        "height": "48px",
        "paddingLeft": "24px",
        "paddingRight": "24px",
      },
      "main": Object {
        "background": "rgba(0,0,0,0)",
        "border": "1px solid #1A99F4",
        "color": "#1A99F4",
      },
      "minWidth": "108px",
      "normal": Object {
        "height": "32px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
      },
    }
  }
  type="button"
>
  Click me
</ButtonWrap>
`;
